java-frontend/src/main/java/org/sonar/java/resolve/ParametrizedTypeJavaType.java

```diff
@@ -64,12 +65,18 @@ public boolean isSubtypeOf(Type superType) {
     if (((JavaType) superType).isTagged(TYPEVAR)) {
       return false;
     }
-    if (erasure().isSubtypeOf(superType.erasure())) {
-      boolean superTypeIsParametrizedJavaType = ((JavaType) superType).isParameterized();
-      if (superTypeIsParametrizedJavaType) {
+    if(erasure() == superType.erasure()) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Change this comparison to use the equals method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1698)